### PR TITLE
Provide clear description of sidecar-injector in README

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
Closes https://github.com/cyberark/sidecar-injector/issues/37

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CyberArk Sidecar Injector
 
+Sidecars are used in Kubernetes to introduce additional features to application pods. Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector** enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). This means when **CyberArk Sidecar Injector** is deployed and enabled in a namespace, any pod created in that namespace, with annotations that indicate a request for injection, will result in automated sidecar injection. **CyberArk Sidecar Injector** provides a selection of available sidecars that are configurable through annotations.
```
Comment (sgnn7) [inline README.md:3]:
```suggestion
Sidecars are used in Kubernetes to introduce additional features to application pods. Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector** enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). This means that when **CyberArk Sidecar Injector** is deployed and enabled in a namespace, any pod created in that namespace with annotations that indicate a request for injection will result in automated sidecar injection. **CyberArk Sidecar Injector** provides a selection of available sidecars that are configurable through annotations with <insert names> being the currently-supported ones.
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+Note that unlike manual injection, automatic injection occurs at the pod-level. You won’t see any change to the deployment itself. Instead you’ll want to check individual pods (via kubectl describe) to see the injected sidecar.
```
Comment (sgnn7) [inline README.md:5]: not a blocker but I think we italicize notes like this usually

File: README.md (hunk 1 above)
Comment (izgeri) [inline README.md:3]:
```suggestion
Sidecars are used in Kubernetes to introduce additional features to application pods.
Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector**
enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook).
This means when **CyberArk Sidecar Injector** is deployed and enabled in a namespace,
any pod created in that namespace with the appropriate annotations will result in automated
sidecar injection. **CyberArk Sidecar Injector** provides support for a selection of available sidecars
that are configurable through annotations.
```
shortens line length, makes minor edits

File: README.md (hunk 2 above)
Comment (sgnn7) [inline README.md:5]:
```suggestion
Note that unlike manual injection, automatic injection occurs at the pod-level. You won’t see any change to the deployment itself. Instead you will want to check individual pods (via `kubectl describe`) to see the injected sidecar.
```

File: README.md [hunk 3]
```diff
@@ -10,7 +14,7 @@ Naming and functionality are still subject to *breaking* changes.
 
 ***
 
-This document shows how to deploy and use the CyberArk Sidecar Injector [MutatingAdmissionWebhook](https://kubernetes.io/docs/admin/admission-controllers/#mutatingadmissionwebhook-beta-in-19) Server which injects sidecar container(s) into a pod prior to persistence of the underlying object.
+This document shows how to deploy and use the CyberArk Sidecar Injector [mutating admission webhook controller](https://kubernetes.io/docs/admin/admission-controllers/#mutatingadmissionwebhook-beta-in-19) which injects sidecar container(s) into a pod prior to persistence of the underlying object.
```
Comment (sgnn7) [inline README.md:17]: Is there a better link than a link to the beta page?

File: README.md (hunk 1 above)
Comment (sgnn7) [reply README.md:3]: non-blocking nit: This paragraph for sure (and maybe even other ones in this doc) may need word wrapping.

File: README.md (hunk 3 above)
Comment (doodlesbykumbi) [reply README.md:17]: yup updated

File: README.md (hunk 1 above)
Comment (doodlesbykumbi) [reply README.md:3]: updated

**Pull request merged**